docs/api.rst

```diff
@@ -299,11 +299,115 @@ Given a higher-order stream, return a new stream that merges all the inner strea
   stream:        -s------t--------->
   join(stream):  ---a---b--4c-5-d6->
 
-concatMap :: (a -> Stream b) -> Stream a -> Stream b
+.. _concatMap:
 
-mergeConcurrently :: int -> Stream (Stream a) -> Stream a
+concatMap
+^^^^^^^^^
+
+.. code-block:: haskell
+
+  concatMap :: (a -> Stream b) -> Stream a -> Stream b
+
+Transform each event in stream into a stream, and then concatenate it onto the 
+end of the resulting stream. Note that f must return a stream.
+
+The mapping function f is applied lazily. That is, f is called only once it is 
+time to concatenate a new stream.
+
+``function f(x) -> Stream``
+
+.. code-block:: javascript
```

Comment:
The best way to format the ascii diagram is to put `::` at the end of the preceding paragraph, and then just indent the ascii diagram rather than using `.. code-block:: javascript` (since it isn't javascript 😄 ).  Like:

```
The mapping function f is applied lazily. That is, f is called only once it is time to concatenate a new stream.::

  stream:                -a----b----c|
  f(a):                   1--2--3|
  f(b):                        1----2----3|
  f(c):                               1-2-3|
  concatMap(f, stream):  -1--2--31----2----31-2-3|
  f called lazily:        ^      ^          ^
```